guildwars2/guild/sync.py

```diff
@@ -290,21 +350,12 @@ def check(m):
                                     except AttributeError:
                                         # role no longer exists - deleted?
                                         pass
-                                try:
-                                    await member.add_roles(
-                                        desiredrole,
-                                        reason="GW2Bot Integration [$guildsync]"
-                                    )
-                                except discord.Forbidden:
-                                    self.log.debug(
-                                        "Permissions error when trying to "
-                                        "give {0} role to {1} user "
-                                        "in {2} server.".format(
-                                            desiredrole.name, member.name,
-                                            guild.name))
-                                except AttributeError:
-                                    # role no longer exists - deleted?
-                                    pass
+                                await self.add_member_to_role(desiredrole, member, guild)
+                            if guildrole:
+                                desiredguildrole = discord.utils.get(
```

Comment:
Style: could probably just overwrite `guildrole` variable, and get that role before the loop. I think so at least, might be tired.